Kitodo/src/main/java/org/kitodo/production/services/data/TaskService.java

```diff
@@ -332,35 +312,68 @@ public TaskDTO convertJSONObjectToDTO(Map<String, Object> jsonObject, boolean re
+        taskInterface.setLocalizedTitle(getLocalizedTitle(taskInterface.getTitle()));
+        taskInterface.setOrdering(TaskTypeField.ORDERING.getIntValue(jsonObject));
+        int taskStatus = TaskTypeField.PROCESSING_STATUS.getIntValue(jsonObject);
+        taskInterface.setProcessingStatus(TaskStatus.getStatusFromValue(taskStatus));
+        taskInterface.setProcessingStatusTitle(Helper.getTranslation(taskInterface.getProcessingStatus().getTitle()));
+        int editType = TaskTypeField.EDIT_TYPE.getIntValue(jsonObject);
+        taskInterface.setEditType(TaskEditType.getTypeFromValue(editType));
+        taskInterface.setEditTypeTitle(Helper.getTranslation(taskInterface.getEditType().getTitle()));
+        try {
+            taskInterface.setProcessingMoment(TaskTypeField.PROCESSING_TIME.getStringValue(jsonObject));
+            taskInterface.setProcessingBeginTime(TaskTypeField.PROCESSING_BEGIN.getStringValue(jsonObject));
+            taskInterface.setProcessingEndTime(TaskTypeField.PROCESSING_END.getStringValue(jsonObject));
+        } catch (ParseException e) {
+            throw new DataException(e);
+        }
+        taskInterface.setCorrection(TaskTypeField.CORRECTION.getBooleanValue(jsonObject));
+        taskInterface.setTypeAutomatic(TaskTypeField.TYPE_AUTOMATIC.getBooleanValue(jsonObject));
+        taskInterface.setTypeMetadata(TaskTypeField.TYPE_METADATA.getBooleanValue(jsonObject));
+        taskInterface.setTypeImagesWrite(TaskTypeField.TYPE_IMAGES_WRITE.getBooleanValue(jsonObject));
+        taskInterface.setTypeImagesRead(TaskTypeField.TYPE_IMAGES_READ.getBooleanValue(jsonObject));
+        taskInterface.setBatchStep(TaskTypeField.BATCH_STEP.getBooleanValue(jsonObject));
+        taskInterface.setRoleIds(convertJSONValuesToList(TaskTypeField.ROLES.getJsonArray(jsonObject)));
+        taskInterface.setRolesSize(TaskTypeField.ROLES.getSizeOfProperty(jsonObject));
+        taskInterface.setCorrectionCommentStatus(TaskTypeField.CORRECTION_COMMENT_STATUS.getIntValue(jsonObject));
+        return taskInterface;
     }
 
     /**
-     * Parses and adds properties related to the project of a task to the taskDTO.
+     * Parses and adds properties related to the project of a task to the taskInterface.
      * 
      * @param jsonObject the jsonObject retrieved from the ElasticSearch index for a task
-     * @param taskDTO the taskDTO
+     * @param taskInterface the taskInterface
      */
-    private void convertTaskProjectFromJsonObjectToDTO(Map<String, Object> jsonObject, TaskDTO taskDTO) throws DataException {
-        ProjectDTO projectDTO = new ProjectDTO();
-        projectDTO.setId(TaskTypeField.PROJECT_ID.getIntValue(jsonObject));
-        projectDTO.setTitle(TaskTypeField.PROJECT_TITLE.getStringValue(jsonObject));
-        taskDTO.setProject(projectDTO);
+    private void convertTaskProjectFromJsonObjectToInterface(Map<String, Object> jsonObject,
```

Comment:
You are naming the variables by the class name. This could be done but it is not so pretty. Instead if naming the variable `taskInterface` a `task` variable from the  `TaskInterface` class should be more readable.

Maybe even do not mention the 'interface' using in method names. Is there no better naming / wording for methods and variables possible where the interface classes are used? 

I'm writing this only on this place but should be changed on all places.

Reply:
These names came about through a RegEx replace from DTO to Interface. I completely agree that variables shouldn't have …interface in their names. I have completely changed this.